src/server/hotkey_collector.h

```diff
@@ -0,0 +1,53 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include <dsn/utility/string_view.h>
+#include <rrdb/rrdb_types.h>
+
+namespace pegasus {
+namespace server {
+
+class hotkey_coarse_data_collector;
+class hotkey_fine_data_collector;
+
+// hotkey_collector is responsible to find the hot keys after the partition
+// was detected to be hot. The two types of hotkey, READ & WRITE, are detected
+// separately.
+class hotkey_collector
+{
+public:
+    // size: the cu size of raw_key/hash_key calculated by `capacity_unit_calculator`
+    void capture_raw_key(const dsn::blob &raw_key, uint64_t size);
+    void capture_hash_key(const dsn::blob &hash_key, uint64_t size);
+    // analyse_data is a periodic task, only valid when _state == collector_state::COARSE
+    // || collector_state::FINE
+    void analyse_data();
+    bool handle_operation(dsn::apps::hotkey_detect_action::type action, std::string &err_hint);
+    // find outlier in both coarse capture and fine capture
+    static int cal_outlier(const std::vector<uint64_t> &data_samples, int threshold);
+    // unify the hash way in both coarse capture and fine capture
+    static int get_bucket_id(dsn::string_view data);
+
```

Comment:
I could not figure out what the functions uage by function definitions and comments. I suggest you remove some function in this pull request, otherwise, you should explain what those functions used for clearly.